### PR TITLE
Cartpage

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds the CartPage into E-commerce app. Upon signing in a user can add items to cart. The item count can be increased/decreased or removed from the cart page.
The components added are-

- `CartCard` at src/components/cart-card
- `PriceDetail` at src/components/price-detail
 
The routes added are-

- `CartPage` at ("/cart") at src/routes/cartpage/CartPage.jsx

The context added are-

- `cart-context` at src/context/cart-context.js
- `user-context` at src/context/user-context.js

The reducer added are-

- `cart-reducer` at src/reducer/cart-reducer.js

[live preview](https://deploy-preview-4--bookstock.netlify.app/)
Would appreciate you feedback and suggestions.

image

image

### CODE REVIEW

File: src/components/cart-card/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const CartCard = ({ image, title, price, includeStock, _id, qty }) => {
+  console.log(qty);
+  const { encodedToken } = useUser();
+  const { dispatch } = useCart();
+
+  const incrementItem = async () => {
+    // dispatch({
+    //   type: "INCREMENT_ITEM",
+    //   payload: { dispatch, _id, encodedToken },
+    // });
```
Comment (krgarima) [inline src/components/cart-card/CartCard.jsx:15]: Remove unnecessary commented code

**Review by krgarima — APPROVED** [review]
I see when the user is not logged in, your cart page looks like this. Fix this.
Screenshot:
asa
Otherwise, the file structure is good. Logic and concepts applied well.
Just remove the extra code before submission.

File: src/routes/cartpage/cartpage.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    gap: 2rem;
+}
+@media (max-width:500px) {
+    .cartpage-container{
+        display: flex;
+        flex-direction: column;
+    }
+    .cartpage-count{
+
+    }
```
Comment (shrushti2000) [inline src/routes/cartpage/cartpage.css]: you can remove this if forgotten to remove.

File: src/components/cart-card/CartCard.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import "./cart-card.css";
+import { useUser } from "../../context/user-context";
+import { useCart } from "../../context/cart-context";
+import axios from "axios";
+
+const CartCard = ({ image, title, price, includeStock, _id, qty }) => {
+  console.log(qty);
```
Comment (shrushti2000) [inline src/components/cart-card/CartCard.jsx]: can remove this console log

**Review by shrushti2000 — APPROVED** [review]
Good job on giving detailed PR description , following good naming conventions and writing clean code. Can remove commented code and console logs .

File: src/components/cart-card/CartCard.jsx (hunk 1 above)
Comment (vishalg8454) [reply src/components/cart-card/CartCard.jsx:15]: - I am still figuring out about this problem of navbar not sticking to bottom, will do in future. 
- The commented actually needs to be debugged in future that's why I have not removed it. Will remove in upcoming release.

File: src/routes/cartpage/cartpage.css (hunk 2 above)
Comment (vishalg8454) [reply src/routes/cartpage/cartpage.css]: sure

File: src/components/cart-card/CartCard.jsx (hunk 3 above)
Comment (vishalg8454) [reply src/components/cart-card/CartCard.jsx]: yeah, sure.

**Pull request merged**